src/GameWindow.java

```diff
@@ -84,9 +84,14 @@ public void mouseExited(MouseEvent event) { // get rid of highlight when mouse l
 	}
 
 	public void actionPerformed(ActionEvent event) { // this is going to contain a lot of stuff too
-		// this is temporary since the board class should be updated to handle column only things
-		int x = (Integer) (((JButton) event.getSource()).getClientProperty("x"));
-		int y = (Integer) (((JButton) event.getSource()).getClientProperty("y"));
-		eventHandler.getBoard().addChip(x, y, eventHandler.getCurrentPlayer());
+		int column = (Integer) (((JButton) event.getSource()).getClientProperty("y"));
+		int row = eventHandler.getBoard().addChip(column, eventHandler.getCurrentPlayer());
+		if (row != -1) {
+			buttonGrid[row+1][column].setBackground(eventHandler.getCurrentPlayer() == 1 ? Color.RED : Color.YELLOW); // reminder: work around board zero limitations by using playerIndex - 1 when getting names
```

Comment:
You don't have to do this but you could disable the button once it is clicked - just a quality of life improvement

Reply:
Yeah sure I'll do that